uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-notification.component.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-env amd */
+
+define([
+  'leave-absences/shared/modules/components'
+], function (components) {
+  components.component('leaveNotification', {
+    bindings: {
+      eventName: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-notification.html';
+    }],
+    controllerAs: 'notification',
+    controller: ['$log', '$rootScope', 'pubSub', 'LeaveRequest', controller]
+  });
+
+  function controller ($log, $rootScope, pubSub, LeaveRequest) {
+    $log.debug('Component: leave-notification');
+
+    var filters = {};
+    var vm = this;
+    vm.count = 0;
+    vm.loading = { count: true };
+
+    (function init () {
+      initializeListeners();
+    })();
+
+    /**
+     * Fetch count of leave requests which matches the filter
+     *
+     * @return {Promise}
+     */
+    function fetchCount () {
+      vm.loading.count = true;
+
+      return LeaveRequest.all(filters)
+        .then(function (leaveRequests) {
+          vm.count = leaveRequests.list.length;
+          vm.loading.count = false;
+        });
+    }
+
+    /**
+     * Initializes the filter and fetches fetches filtered leave requests
+     * @param {Object} e - Event object
+     * @param {Object} filtersData - Filters
+     * @return {Promise}
+     */
+    function initializeFilters (e, filtersData) {
+      filters = filtersData;
+
+      return fetchCount();
+    }
+
+    /**
+     * Initializes the event listeners
+     */
+    function initializeListeners () {
+      $rootScope.$on('WaitingApproval:: Initialize Filters::' + vm.eventName, initializeFilters);
```

Comment:
I think this is wrong for two reasons:

1) If the `leave-notification-badge` should be generic, then you can't have a reference to "waiting approval", because that's a detail implementation that the component should know nothing about
2) Why the component needs to have this event anyway? If you give it the filters and the event name, why can'it it fetch the count immediately on `init`?